torch/lib/c10d/TCPStore.cpp

```diff
@@ -159,6 +162,20 @@ void TCPStoreDaemon::setHandler(int socket) {
   wakeupWaitingClients(key);
 }
 
+void TCPStoreDaemon::compareSetHandler(int socket) {
+  std::string key = tcputil::recvString(socket);
+  std::vector<uint8_t> currentValue = tcputil::recvVector<uint8_t>(socket);
+  std::vector<uint8_t> newValue = tcputil::recvVector<uint8_t>(socket);
+
+  // sets value to new value if the key exists and the existing value is the old value
+  if (tcpStore_.find(key) != tcpStore_.end() && tcpStore_.at(key) == currentValue) {
```

Comment:
Also, depending on how we handle the above case, you might be able to save the iterator returned by find and use that when comparing to currentValue instead of indexing into the map a second time with `tcpStore_.at(key)`.

Reply:
> What is the expected behavior if the key does not exist in the map? Currently if the key doesn't exist, I believe the second condition will result in an out of range exception

Am I overseeing something? Wouldn't the iterator returned from `tcpStore_.find(key)` equal to `tcpStore_.end()` if the key was not found? In such case the second condition wouldn't be checked anyways.

Reply:
Yep, if the key is not found, it doesn't hit the 2nd conditional. Also changed it to use an iterator as mentioned.

Reply:
Ah yeah thanks for pointing that out - missed that for some reason :)